mux_test.go

```diff
@@ -35,8 +35,10 @@ type routeTest struct {
 	path           string            // the expected path of the match
 	pathTemplate   string            // the expected path template to match
 	hostTemplate   string            // the expected host template to match
+	additionalPath string            // additional path to add to route (and parse) during test
```

Comment:
I just added an alternative test - and I didn't see this last comment before I pushed it.

Let me know if you're happy with the alternative which I think is probably more reusable than it previously was, or if you'd like me to just test the parsing separately and remove the additional fields.